config/src/main/scala/bloop/config/Config.scala

```diff
@@ -63,45 +59,68 @@ object Config {
     private[bloop] val empty: Scala = Scala("", "", "", Array(), Array())
   }
 
-  sealed abstract class Platform(val name: String)
+  sealed abstract class Platform(val name: String) {
+    type Config <: PlatformConfig
+    def config: Config
+  }
+
   object Platform {
-    private[bloop] val default: Platform = JVM
+    private[bloop] val default: Platform = Jvm(JvmConfig.empty)
+
+    object Js { val name: String = "js" }
+    case class Js(override val config: JsConfig) extends Platform(Js.name) {
+      type Config = JsConfig
+    }
+
+    object Jvm { val name: String = "jvm" }
+    case class Jvm(override val config: JvmConfig) extends Platform(Jvm.name) {
+      type Config = JvmConfig
+    }
 
-    case object JS extends Platform("js")
-    case object JVM extends Platform("jvm")
-    case object Native extends Platform("native")
+    object Native { val name: String = "native" }
+    case class Native(override val config: NativeConfig) extends Platform(Native.name) {
+      type Config = NativeConfig
+    }
 
-    final val All: List[String] = List(JVM.name, JS.name, Native.name)
+    final val All: List[String] = List(Jvm.name, Js.name, Native.name)
   }
 
+  sealed trait PlatformConfig
+
+  case class JvmConfig(home: Option[Path], options: List[String]) extends PlatformConfig
+  object JvmConfig { private[bloop] val empty = JvmConfig(None, Nil) }
+
+  case class JsConfig(toolchainClasspath: List[Path]) extends PlatformConfig
+  object JsConfig { private[bloop] val empty: JsConfig = JsConfig(Nil) }
+
   /**
-   * Configures how to start and use the Scala Native toolchain, if needed.
+   * Represents the native platform and all the options it takes.
+   *
    * For the description of these fields, see:
    * http://static.javadoc.io/org.scala-native/tools_2.10/0.3.7/index.html#scala.scalanative.build.Config
+   *
+   * The only field that has been replaced for user-friendliness is `targetTriple` by `platform`.
    */
   case class NativeConfig(
-      toolchainClasspath: Array[Path],
       gc: String,
-      clang: Path,
-      clangPP: Path,
-      linkingOptions: Array[String],
-      compileOptions: Array[String],
-      targetTriple: String,
+      platform: String,
```

Comment:
Are you sure about renaming `targetTriple` to `platform`? We already have `platform` as in `JVM` vs `JS` vs `Native` so this is a bit confusing. `target` is the standard way to refer to architecture/OS/ABI when compiling native code. OTOH `target` may also refer to the path where compiled binaries are written, so maybe something like `targetPlatform`?

Reply:
You're right, I didn't notice we also had another `platform`. I just pushed a commit with the fix :wink: